src/ReactTestWrapper.js

```diff
@@ -90,4 +90,8 @@ export default class ReactTestWrapper extends TestWrapper {
   hasRef (ref) {
     return !!this.wrapper.instance().refs[ref]
   }
+
+  type () {
+    throw new Error('full DOM rendering does not support `type` yet')
+  }
```

Comment:
I wonder if we can use the (display) name (or something else) of a component somehow. I really hope we can implement this assertion for each type of rendering.